src/librustc_passes/check_attr.rs

```diff
@@ -4,138 +4,28 @@
 //! conflicts between multiple such attributes attached to the same
 //! item.
 
-use crate::hir::map::Map;
-use crate::ty::query::Providers;
-use crate::ty::TyCtxt;
+use rustc::hir::check_attr::{MethodKind, Target};
+use rustc::hir::map::Map;
+use rustc::ty::query::Providers;
+use rustc::ty::TyCtxt;
 
 use rustc_errors::struct_span_err;
 use rustc_hir as hir;
 use rustc_hir::def_id::DefId;
 use rustc_hir::intravisit::{self, NestedVisitorMap, Visitor};
 use rustc_hir::DUMMY_HIR_ID;
-use rustc_hir::{self, HirId, Item, ItemKind, TraitItem, TraitItemKind};
+use rustc_hir::{self, HirId, Item, ItemKind, TraitItem};
 use rustc_session::lint::builtin::UNUSED_ATTRIBUTES;
 use rustc_span::symbol::sym;
 use rustc_span::Span;
 use syntax::ast::Attribute;
 use syntax::attr;
 
-use std::fmt::{self, Display};
-
-#[derive(Copy, Clone, PartialEq)]
-pub(crate) enum MethodKind {
-    Trait { body: bool },
-    Inherent,
-}
-
-#[derive(Copy, Clone, PartialEq)]
-pub(crate) enum Target {
-    ExternCrate,
-    Use,
-    Static,
-    Const,
-    Fn,
-    Closure,
-    Mod,
-    ForeignMod,
-    GlobalAsm,
-    TyAlias,
-    OpaqueTy,
-    Enum,
-    Struct,
-    Union,
-    Trait,
-    TraitAlias,
-    Impl,
-    Expression,
-    Statement,
-    AssocConst,
-    Method(MethodKind),
-    AssocTy,
-    ForeignFn,
-    ForeignStatic,
-    ForeignTy,
+pub(crate) trait TargetExt {
+    fn from_impl_item<'tcx>(tcx: TyCtxt<'tcx>, impl_item: &hir::ImplItem<'_>) -> Target;
```

Comment:
I think this method can just stay in `rustc`.

Reply:
This method is only used in this file. Since `rustc::hir::check_attr` is bound to disappear in #68554, I think I will make it a free function.